src/main/kotlin/pl/ludek/poolserver/plugins/SimpleModbusRTURelay.kt

```diff
@@ -74,13 +76,13 @@ class SimpleModbusRTURelay {
     fun getStateAllRelay(): RelayState {
         val dataState = getStateRelay()
         if (dataState == null){
-        return RelayState(Array<Boolean>(9){true})
+        return RelayState(Array<Boolean>(8){false}, error)
         }else{
-            val array:Array<Boolean> = Array<Boolean>(9){false}
-            for (i in 0..7){
+            val array= Array<Boolean>(8){false}
+            for (i in 0..array.size-1){
```

Comment:
You can use `for (i in array.indices){` or `for (i in 0 until array.size){` to avoid `array.size-1`